### ISSUE TITLE
Plotting labels instead of markers

### ISSUE THREAD

**Issue by urisohn** [aroneklund/beeswarm#22]
Been using beeswarm for a while (e.g., in our blog Data Colada). Now using it for a new pkg that creates plots for reporting results from psychology experiments, and one of the plots we wanted to offer was the beeswarm plot, but using text-labels instead of markers. 
Like. Imagine with data at the State level in the US and every 'marker' is the two letter abbreviation for the state (AZ, AK, etc.)

![Karmali   Kawakami - stimulus beeswarm](https://github.com/user-attachments/assets/cad88c56-ca4c-4626-a4e2-8433c759f7a2)
You can see an example inm the uploaded plot. I ended up creating a beeswarm, without plotting, using the exprted values, and then using text() to populate it, but maybe it could be built into the package as an option. Just a suggestion.

**Comment by aroneklund** [comment]
Hmm.  That's not a bad idea. 

I guess you could create a simple version with something like this (which I think is what you are describing):
```
out <- beeswarm(state.area, log = TRUE, spacing = 4, pch = NA)
text(out$x, out$y, labels = state.abb)
```

Here I set `spacing = 4` based on trial and error.  I think this works pretty well with state abbreviations, since they are all around the same size.

Are you proposing a more automatic approach, such as taking the size of each individual label into account when doing the layout?  (maybe similar to #5 but with text instead of points?)

**Comment by urisohn** [comment]
Yes, that's basically what i did, but i think some automatic spacing would be needed to prevent overstepping too much
i find it hard to navigate the mapping between text and absolute dimensions so could be tricky

The hacky way I did was to compute the average length of the labels and convert that to the horizontal spacing.
LIke this

```
 stimulus.length = mean(nchar(ms$stimulus))
  dot.spacing = stimulus.length/3 + 2
```

where ms$stimulus contains the labels you see, and then i make spacing=dot.spacing inside the beeswarm call
